### PR TITLE
[Style] Update stepper

### LINKED ISSUE #9550: ♻️ Update the progress stepper
## ✨ Summary

[Visit the Figma file](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=5-26&mode=design) • Contact @substrae with design questions

This ticket aims to update the style of the progress stepper's iconography to make things clearer and easier to interpret. The icons in the progress stepper as they stand are extremely small and compete visually with the background colors in a way that makes them more noise than helpful. As a result, we're opting to remove icons from steps unless they are one of the three key states:
* Active
* Completed
* In error

```[tasklist]
### ✅ Requirements
- [ ] Remove icons from the default step state
- [ ] Update the active state to match
- [ ] Update the error state to match
- [ ] Update the success state to match
- [ ] Update the default color (circle and stem) to `gray.80`
- [ ] Update the dark mode default color (circle and stem) to `background.darker.80`
- [ ] Triple check contrast values
- [ ] Triple check dark mode
- [ ] Triple check the IAP theme
``` 

```[tasklist]
### 🔗 Related
- [ ] #9551 
``` 

## 🔍 Details

<details>
  <summary>📸 Mid-completion</summary>
  
</details> 

<details>
  <summary>📸 With error</summary>
  
</details> 

<details>
  <summary>📸 Complete</summary>
  
</details>

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #9550 

## 👋 Introduction

Makes some minor tweaks to the stepper component that appears within applications.

## 🕵️ Details

We used some hardcoded values since we could not use opacity easily. @substrae provided the values used.

- Light: `#C5C6C7`
- Dark: `#7C89A3`

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Login as applicant `applicant@test.com`
3. Start an application
4. Confirm the stepper matches [the mockup](https://www.figma.com/file/guHeIIh8dqFVCks310Wv0G/Style-library?type=design&node-id=5-26&mode=design&t=ySGCbzrNufVcF3tN-0)

## 📸 Screenshot

Screenshot 2024-03-04 134945

### CODE REVIEW

File: packages/ui/src/components/Stepper/utils.ts [hunk 1]
```diff
@@ -36,7 +33,8 @@ export const linkStyleMap = new Map<StepState, Record<string, string>>([
       "data-h2-background-color": `
         base:all:children[.Step__Icon](primary.light)
         base:iap:children[.Step__Icon](primary)
-        base:all:children[.Step__Tail](gray.light)
+        base:children[.Step__Tail](#C5C6C7)
+        base:dark:children[.Step__Tail](#7C89A3)
```
Comment (petertgiles) [inline packages/ui/src/components/Stepper/utils.ts:37]: Could you add a comment about how these colours are calculated or what they represent?

Comment (esizer) [reply packages/ui/src/components/Stepper/utils.ts:37]: @substrae can you help out here? I actually don't know 😅 I can add the comment if you just provide the information Peter is looking for.

File: packages/ui/src/components/Stepper/Step.tsx
Comment (petertgiles) [inline packages/ui/src/components/Stepper/Step.tsx:1]: I don't know if this is even possible with heroicons but it looks like the check and x are supposed to be transparent.
image

**Review by petertgiles — CHANGES_REQUESTED** [review]
Looks really nice!  I left two questions to check on.

File: packages/ui/src/components/Stepper/utils.ts (hunk 1 above)
Comment (substrae) [reply packages/ui/src/components/Stepper/utils.ts:37]: Sure thing - Figma allows for multi-layered background colors, so the design leveraged the page background along with an opacity of the respective light and dark colors. Using that opacity in practice in code created transparent semi-overlapping elements which we didn't want, so we color picked the merged colors from the Figma file to create a solid version of what the transparency in the design generated.

File: packages/ui/src/components/Stepper/Step.tsx
Comment (esizer) [reply packages/ui/src/components/Stepper/Step.tsx:1]: Using the mini versions, should we swap for the normal ones?

Comment (esizer) [reply packages/ui/src/components/Stepper/Step.tsx:1]: Updated the colour in [update icon colour to use background](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9628/commits/1be290567ef1c6c33e9f8b953ec24685215cec7f)

File: packages/ui/src/components/Stepper/utils.ts (hunk 1 above)
Comment (esizer) [reply packages/ui/src/components/Stepper/utils.ts:50]: okay, hopefully my paraphrased explanation is accurate and makes sense for devs 😛 

[document custom hex colour codes](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9628/commits/1c4c91adc0d40030c7734360eefe43b4af58f9d5)

File: packages/ui/src/components/Stepper/Step.tsx
Comment (substrae) [reply packages/ui/src/components/Stepper/Step.tsx:1]: Weiiiiird. Nah, keep'em I guess. They look so different 🤔

File: packages/ui/src/components/Stepper/utils.ts (hunk 1 above)
Comment (substrae) [reply packages/ui/src/components/Stepper/utils.ts:50]: Makes sense to me! But I guess I'm the opposite of the target audience 🤣

**Review by petertgiles — APPROVED** [review]

**Pull request merged**